From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Calum Lind <calumlind+deluge@gmail.com>
Date: Tue, 3 Mar 2026 21:07:17 +0000
Subject: [PATCH] feat(plugins): replace common.py pkg_resources with importlib

Return a str instead of Path since most resource usage requires a str
filepath e.g. GTK builder calls.
---
 deluge/plugins/AutoAdd/deluge_autoadd/common.py        | 10 ++++------
 deluge/plugins/Blocklist/deluge_blocklist/common.py    |  8 +++-----
 deluge/plugins/Execute/deluge_execute/common.py        |  8 +++-----
 deluge/plugins/Extractor/deluge_extractor/common.py    |  8 +++-----
 deluge/plugins/Label/deluge_label/common.py            |  8 +++-----
 .../Notifications/deluge_notifications/common.py       |  7 +++----
 deluge/plugins/Scheduler/deluge_scheduler/common.py    |  8 +++-----
 deluge/plugins/Stats/deluge_stats/common.py            |  8 +++-----
 deluge/plugins/Toggle/deluge_toggle/common.py          |  8 +++-----
 deluge/plugins/WebUi/deluge_webui/common.py            |  8 +++-----
 deluge/scripts/create_plugin.py                        | 10 +++-------
 deluge/ui/web/server.py                                |  2 +-
 12 files changed, 35 insertions(+), 58 deletions(-)

diff --git a/deluge/plugins/AutoAdd/deluge_autoadd/common.py b/deluge/plugins/AutoAdd/deluge_autoadd/common.py
index 6a790cbd598e..ea20404750e9 100644
--- a/deluge/plugins/AutoAdd/deluge_autoadd/common.py
+++ b/deluge/plugins/AutoAdd/deluge_autoadd/common.py
@@ -11,11 +11,9 @@
 # See LICENSE for more details.
 #
 
-import os.path
-
-from pkg_resources import resource_filename
+from importlib.resources import files
 
 
-def get_resource(filename, subdir=False):
-    folder = os.path.join('data', 'autoadd_options') if subdir else 'data'
-    return resource_filename(__package__, os.path.join(folder, filename))
+def get_resource(filename: str, subdir: bool = False) -> str:
+    folder = 'data/autoadd_options' if subdir else 'data'
+    return str(files(__package__) / folder / filename)
diff --git a/deluge/plugins/Blocklist/deluge_blocklist/common.py b/deluge/plugins/Blocklist/deluge_blocklist/common.py
index f1059413de0a..0af864b72831 100644
--- a/deluge/plugins/Blocklist/deluge_blocklist/common.py
+++ b/deluge/plugins/Blocklist/deluge_blocklist/common.py
@@ -11,15 +11,13 @@
 # See LICENSE for more details.
 #
 
-import os.path
 from functools import wraps
+from importlib.resources import files
 from sys import exc_info
 
-from pkg_resources import resource_filename
 
-
-def get_resource(filename):
-    return resource_filename(__package__, os.path.join('data', filename))
+def get_resource(filename: str) -> str:
+    return str(files(__package__) / 'data' / filename)
 
 
 def raises_errors_as(error):
diff --git a/deluge/plugins/Execute/deluge_execute/common.py b/deluge/plugins/Execute/deluge_execute/common.py
index eb47f13985f7..12de6b65d772 100644
--- a/deluge/plugins/Execute/deluge_execute/common.py
+++ b/deluge/plugins/Execute/deluge_execute/common.py
@@ -11,10 +11,8 @@
 # See LICENSE for more details.
 #
 
-import os.path
-
-from pkg_resources import resource_filename
+from importlib.resources import files
 
 
-def get_resource(filename):
-    return resource_filename(__package__, os.path.join('data', filename))
+def get_resource(filename: str) -> str:
+    return str(files(__package__) / 'data' / filename)
diff --git a/deluge/plugins/Extractor/deluge_extractor/common.py b/deluge/plugins/Extractor/deluge_extractor/common.py
index eb47f13985f7..12de6b65d772 100644
--- a/deluge/plugins/Extractor/deluge_extractor/common.py
+++ b/deluge/plugins/Extractor/deluge_extractor/common.py
@@ -11,10 +11,8 @@
 # See LICENSE for more details.
 #
 
-import os.path
-
-from pkg_resources import resource_filename
+from importlib.resources import files
 
 
-def get_resource(filename):
-    return resource_filename(__package__, os.path.join('data', filename))
+def get_resource(filename: str) -> str:
+    return str(files(__package__) / 'data' / filename)
diff --git a/deluge/plugins/Label/deluge_label/common.py b/deluge/plugins/Label/deluge_label/common.py
index eb47f13985f7..12de6b65d772 100644
--- a/deluge/plugins/Label/deluge_label/common.py
+++ b/deluge/plugins/Label/deluge_label/common.py
@@ -11,10 +11,8 @@
 # See LICENSE for more details.
 #
 
-import os.path
-
-from pkg_resources import resource_filename
+from importlib.resources import files
 
 
-def get_resource(filename):
-    return resource_filename(__package__, os.path.join('data', filename))
+def get_resource(filename: str) -> str:
+    return str(files(__package__) / 'data' / filename)
diff --git a/deluge/plugins/Notifications/deluge_notifications/common.py b/deluge/plugins/Notifications/deluge_notifications/common.py
index 9993f5ce4132..933aea0f7724 100644
--- a/deluge/plugins/Notifications/deluge_notifications/common.py
+++ b/deluge/plugins/Notifications/deluge_notifications/common.py
@@ -12,19 +12,18 @@
 #
 
 import logging
-import os.path
+from importlib.resources import files
 
-from pkg_resources import resource_filename
 from twisted.internet import defer
 
 from deluge import component
 from deluge.event import known_events
 
 log = logging.getLogger(__name__)
 
 
-def get_resource(filename):
-    return resource_filename(__package__, os.path.join('data', filename))
+def get_resource(filename: str) -> str:
+    return str(files(__package__) / 'data' / filename)
 
 
 class CustomNotifications:
diff --git a/deluge/plugins/Scheduler/deluge_scheduler/common.py b/deluge/plugins/Scheduler/deluge_scheduler/common.py
index eb47f13985f7..12de6b65d772 100644
--- a/deluge/plugins/Scheduler/deluge_scheduler/common.py
+++ b/deluge/plugins/Scheduler/deluge_scheduler/common.py
@@ -11,10 +11,8 @@
 # See LICENSE for more details.
 #
 
-import os.path
-
-from pkg_resources import resource_filename
+from importlib.resources import files
 
 
-def get_resource(filename):
-    return resource_filename(__package__, os.path.join('data', filename))
+def get_resource(filename: str) -> str:
+    return str(files(__package__) / 'data' / filename)
diff --git a/deluge/plugins/Stats/deluge_stats/common.py b/deluge/plugins/Stats/deluge_stats/common.py
index eb47f13985f7..12de6b65d772 100644
--- a/deluge/plugins/Stats/deluge_stats/common.py
+++ b/deluge/plugins/Stats/deluge_stats/common.py
@@ -11,10 +11,8 @@
 # See LICENSE for more details.
 #
 
-import os.path
-
-from pkg_resources import resource_filename
+from importlib.resources import files
 
 
-def get_resource(filename):
-    return resource_filename(__package__, os.path.join('data', filename))
+def get_resource(filename: str) -> str:
+    return str(files(__package__) / 'data' / filename)
diff --git a/deluge/plugins/Toggle/deluge_toggle/common.py b/deluge/plugins/Toggle/deluge_toggle/common.py
index eb47f13985f7..12de6b65d772 100644
--- a/deluge/plugins/Toggle/deluge_toggle/common.py
+++ b/deluge/plugins/Toggle/deluge_toggle/common.py
@@ -11,10 +11,8 @@
 # See LICENSE for more details.
 #
 
-import os.path
-
-from pkg_resources import resource_filename
+from importlib.resources import files
 
 
-def get_resource(filename):
-    return resource_filename(__package__, os.path.join('data', filename))
+def get_resource(filename: str) -> str:
+    return str(files(__package__) / 'data' / filename)
diff --git a/deluge/plugins/WebUi/deluge_webui/common.py b/deluge/plugins/WebUi/deluge_webui/common.py
index eb47f13985f7..12de6b65d772 100644
--- a/deluge/plugins/WebUi/deluge_webui/common.py
+++ b/deluge/plugins/WebUi/deluge_webui/common.py
@@ -11,10 +11,8 @@
 # See LICENSE for more details.
 #
 
-import os.path
-
-from pkg_resources import resource_filename
+from importlib.resources import files
 
 
-def get_resource(filename):
-    return resource_filename(__package__, os.path.join('data', filename))
+def get_resource(filename: str) -> str:
+    return str(files(__package__) / 'data' / filename)
diff --git a/deluge/scripts/create_plugin.py b/deluge/scripts/create_plugin.py
index 7383661ea162..a77a93d4166e 100644
--- a/deluge/scripts/create_plugin.py
+++ b/deluge/scripts/create_plugin.py
@@ -222,15 +222,11 @@ def __init__(self, plugin_name):
 )
 """
 
-COMMON = """from __future__ import unicode_literals
-
-import os.path
-
-from pkg_resources import resource_filename
+COMMON = """from importlib.resources import files
 
 
-def get_resource(filename):
-    return resource_filename(__package__, os.path.join('data', filename))
+def get_resource(filename: str) -> str:
+    return str(files(__package__) / 'data' / filename)
 """
 
 GTK3UI = """from __future__ import unicode_literals
diff --git a/deluge/ui/web/server.py b/deluge/ui/web/server.py
index e93d4f6f39f7..93ff6995a7d0 100644
--- a/deluge/ui/web/server.py
+++ b/deluge/ui/web/server.py
@@ -335,7 +335,7 @@ def add_script(self, path, filepath, script_type=None):
         if script_type not in ('dev', 'debug', 'normal'):
             script_type = 'normal'
 
-        self.__scripts[script_type]['scripts'][path] = filepath
+        self.__scripts[script_type]['scripts'][path] = str(filepath)
         self.__scripts[script_type]['order'].append(path)
         if not os.path.isfile(filepath):
             self.__scripts[script_type]['files_exist'] = False
